### ISSUE TITLE
Mac update relaunch doesn't launch new version of Brave

### ISSUE THREAD

**Issue by RyanJarv** [brave/brave-browser#977]
## Description 
When updating from 0.54.4 to 0.55.1 clicking relaunch after the install closes the current window but doesn't launch the new version.

## Steps to Reproduce

   1. Install 0.54.4
   2. Navigate to chrome://chrome
   3. Click relaunch after update is done downloading/installing

## Actual result:
Browser is closed and we don't see the new version open.

Brave can be launched again manually and will be on the right version though.

## Expected result:

Clicking relaunch open's the updated version.

## Reproduces how often: 
Always

## Brave version (about:brave info) 
0.54.4 -> 0.55.1

### Reproducible on current release:
Yes

Seeing this in Console around the time of the update:
> error    <redacted time>    taskgated-helper    com.brave.Browser.dev: Unsatisfied entitlements: com.apple.application-identifier

**Comment by petemill** [comment]
Same happened with 0.55.1 -> 0.55.2 for @bsclifton and me

**Comment by w0ts0n** [comment]
Just wanted to note, it did work for me. Also on MacOS.

**Comment by aekeus** [comment]
Did not work for me 0.55.1 -> 0.55.2 (macOS)

**Comment by bbondy** [comment]
Did not work for me 0.55.1 -> 0.55.2 (macOS)

**Comment by bsclifton** [comment]
Didn't for for me again ☹️ 0.55.2 => 0.55.3

**Comment by bsclifton** [comment]
With this last update (0.55.3 => 0.55.4), it looks like it tried to relaunch? but then quit really quick (the icon blipped). Had to manually relaunch 🙀

**Comment by yrliou** [comment]
Did not work for me 0.55.4 -> 0.55.5 (macOS)

**Comment by ltilve** [comment]
I was also able to still reproduce the issue when updating from the latest version.

Something that seems it could be related with the issue, is an error message related with the signing of the package that we had been able to see when starting the browser from the command line:

<img width="943" alt="chrome-brave-dev" src="https://user-images.githubusercontent.com/2891281/46621238-406f5000-cb27-11e8-8240-a2c7985ad0ea.png">

**Comment by ltilve** [comment]
I have been using directly the call to `chrome://restart` after installing the Mac package, which is always properly relaunching the browser when triggered before updating, and also after the updated version was started afterwards; but which has the very same issue if it's manually triggered at the moment the update download is completed.

![reload](https://user-images.githubusercontent.com/2891281/46702413-48acb580-cc23-11e8-9776-da7140433c22.png)

So the issue is apparently not coming from anything specific of the `relaunch` action on the `settings/help` or any restart, but just when the bundle has changed.

**Comment by bbondy** [comment]
The task involves some devops work for @mbacchi and we won't be able to get to it in time for 0.55.x So moving it as a priority/p2 in 1.x for asap attention when we get past this release. We'll uplift it into the next milestone or a hotfix.

**Comment by kjozwiak** [comment]
@bbondy @rebron @bsclifton we should really get this fixed. Every time we release an update, we have some complaints from the community and internally. Users who don't know this issue exists usually think that thier update failed. Added this into the `General` so we can discuss during our triage meetings.

**Comment by w0ts0n** [comment]
We implemented/setup a new omaha system to fix the slow downloads issue. Once we are done testing and are sure it's fixed, we can run an update on the staging sparkle/omaha which may fix this issue. We should wait on Simon to get back from PTO too. 

We know this has to be fixed, but I'm not calling this urgent. it's an annoying bug, but not a blocker.

**Comment by simonhong** [comment]
Also, need to find how to update test on test server and how to build image.

**Comment by w0ts0n** [comment]
FYI, team will look into this next week.

**Comment by simonhong** [comment]
I think sparkle isn't compatible with chromium's relaunch procedure.
Instead, we should use sparkle's own install and relaunch mechanism.

**Comment by LaurenWags** [comment]
Verified passed on `macOS x64 10.12.6` when updating from 0.58.12 -> 0.58.14 on `test` channel.

Brave | 0.58.14 Chromium: 71.0.3578.98 (Official Build) (64-bit)
-- | --
Revision | 15234034d19b85dcd9a03b164ae89d04145d8368-refs/branch-heads/3578@{#897}
OS | Mac OS X

cc @kjozwiak for verification on his mac

**Comment by michaelkruglos** [comment]
Doesn't work in the recent version

**Comment by kjozwiak** [comment]
> Doesn't work in the recent version
> When trying in settings:
> 
> > Update failed (error: 12)
> > Error details:
> > An error occurred while relaunching Brave Browser, but the new version will be available next time you run Brave Browser.

Assuming you saw the above via `brave://settings/help`? Did Brave eventually update after restarting? BTW, this is a different issue compared to the above one so if you're still having issues updating, please create a new issue via the following:

* https://github.com/brave/brave-browser/issues/new?assignees=&labels=OS%2FDesktop&template=desktop.md&title=
